### PR TITLE
Enable seq_pool op to accept len 0 input

### PR DESCRIPTION
**Pull request by kuke**
Related to https://github.com/PaddlePaddle/Paddle/pull/16935

The seq_pool op has not support sequence input with length 0 in last pull request, because it is more complicated and needs more consideration.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_seq_pool.py [hunk 1]
```diff
@@ -214,7 +255,7 @@ def compute(self, x, offset, out):
 
 class TestSeqFirstPool2D(TestSeqAvgPool2D):
```
Comment (phlrain) [inline python/paddle/fluid/tests/unittests/test_seq_pool.py:325]: lod length = 0的情况好像没有考虑到

File: python/paddle/fluid/tests/unittests/test_seq_pool.py [hunk 2]
```diff
@@ -167,31 +196,43 @@ def test_check_grad(self):
...
                                (-1, 3 * 11))
             out[i] = np.reshape(np.amax(sub_x, axis=0), (3, 11))
 
 
+class TestSeqMaxPool2DLen0(TestSeqMaxPool2D):
+    def set_lod(self):
+        self.lod = [[0, 3, 10, 0]]
+
+
 class TestSeqMaxPool2DInference(TestSeqMaxPool2D):
```
Comment (phlrain) [inline python/paddle/fluid/tests/unittests/test_seq_pool.py:286]: lod length = 0 的情况

Comment (kuke) [reply python/paddle/fluid/tests/unittests/test_seq_pool.py:286]: Done

File: python/paddle/fluid/tests/unittests/test_seq_pool.py (hunk 1 above)
Comment (kuke) [reply python/paddle/fluid/tests/unittests/test_seq_pool.py:325]: DONE

File: python/paddle/fluid/tests/unittests/test_seq_pool.py [hunk 3]
```diff
@@ -20,40 +20,52 @@
...
 class TestSeqAvgPool(OpTest):
+    def set_lod(self):
+        return [[11]]
+
     def set_data(self):
         self.op_type = 'sequence_pool'
         # one level, batch size is 4
         x = np.random.uniform(0.1, 1, [11, 23]).astype('float32')
-        lod = [[11]]
+        lod = self.set_lod()
```
Comment (phlrain) [inline python/paddle/fluid/tests/unittests/test_seq_pool.py:58]: batch size = 1， 和上面注释的4 不一致

Comment (kuke) [reply python/paddle/fluid/tests/unittests/test_seq_pool.py:58]: Removed this line

**Review by phlrain — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/sequence_ops/sequence_pool_op.cc [hunk 4]
```diff
@@ -37,6 +37,7 @@ class SequencePoolOp : public framework::OperatorWithKernel {
   }
 };
 
+template <typename T>
 class SequencePoolOpMaker : public framework::OpProtoAndCheckerMaker {
```
Comment (chengduoZH) [inline paddle/fluid/operators/sequence_ops/sequence_pool_op.cc]: Please don't use `template ` here, it is an error usage.

Comment (kuke) [reply paddle/fluid/operators/sequence_ops/sequence_pool_op.cc]: DONE

**Review by chengduoZH — APPROVED** [review]
LG API.spec

**Review by xsrobin — APPROVED** [review]
LGTM

**Pull request merged**